### PR TITLE
Implement partial responses support in MercuryManager

### PR DESCRIPTION
**Pull request by tobiasguyer**
The mercury manager failed to merge truncated gzip messages. I rewrote the response decoding to be compüatible with trucated messages. I changed the structure of the response handling slightly, so it would be able to handle incoming asynchronous messages.

### CODE REVIEW

**Comment by feelfreelinux** [comment]
Thanks for your contribution, will try to test it out soon :) What kind of messages from spotify are sent in gzip form - probably missed out on it during the implementation

File: cspot/src/MercurySession.cpp [hunk 1]
```diff
@@ -174,19 +174,25 @@ void MercurySession::handlePacket() {
...
 
-      auto uri = std::string(response.mercuryHeader.uri);
-      if (this->subscriptions.count(uri) > 0) {
-        this->subscriptions[uri](response);
+      if (response.first) {
+        auto partial = this->partials.find(response.second);
+        auto uri = std::string(partial->second.mercuryHeader.uri);
+        if (this->subscriptions.count(uri) > 0) {
+          this->subscriptions[uri](partial->second);
+        }
```
Comment (feelfreelinux) [inline cspot/src/MercurySession.cpp:196]: Shouldn't we also clear the partials map in the subscription response code?

**Comment by feelfreelinux** [comment]
Took the liberty of doing some minor fixes :) Merging into the develop branch, thanks a lot for your contribution!

**Comment by tobiasguyer** [comment]
thanks a lot for the changes. I'm not exactly sure if those responses are gzipped, but at least they share the structure and there are no more problems with long playlists :)

**Pull request merged**